### PR TITLE
Update core.py

### PR DESCRIPTION
**Pull request by 20161116**
### Checklist
- [x] I've tested that my changes are compatible with the latest version of Tensorflow.
- [x] I've read the [Contribution Guidelines](https://github.com/tensorlayer/tensorlayer/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### CODE REVIEW

**Comment by zsdonghao** [comment]
hi @20161116  thank you for the PR, have you test it on `tutorial_mnist_float16.py` ?

**Comment by 20161116** [comment]
I just tested it for two iterations, it took too much time. 
Epoch 1 of 500 took 301.238368s
   train loss: 0.588028
   train acc: 0.878385
   val loss: 0.552224
   val acc: 0.891827
Epoch 2 of 500 took 302.669358s
   train loss: 0.284668
   train acc: 0.935697
   val loss: 0.257953
   val acc: 0.944311
